src/rust/lib/frp/src/nodes.rs

```diff
@@ -113,119 +124,148 @@ impl Network {
+    /// the same output data type. Please note that `any_mut` can be used to create recursive FRP
+    /// networks by creating an empty any and using the `attach` method to attach new streams to
+    /// it. When a recursive network is created, `any_mut` breaks the cycle. After passing the first
     /// event, no more events will be passed till the end of the current FRP network resolution.
-    pub fn gather<T:Data>(&self, label:Label) -> Merge<T> {
-        self.register_raw(OwnedMerge::new(label))
+    pub fn any_mut<T:Data>(&self, label:Label) -> Any<T> {
+        self.register_raw(OwnedAny::new(label))
     }
 
-    /// Merges multiple input streams into a single output stream. All input streams have to share
+    /// Anys multiple input streams into a single output stream. All input streams have to share
     /// the same output data type.
-    pub fn merge<T1,T2,T:Data>(&self, label:Label, t1:&T1, t2:&T2) -> Stream<T>
+    pub fn any<T1,T2,T:Data>(&self, label:Label, t1:&T1, t2:&T2) -> Stream<T>
     where T1:EventOutput<Output=T>, T2:EventOutput<Output=T> {
-        self.register(OwnedMerge::new2(label,t1,t2))
+        self.register(OwnedAny::new2(label,t1,t2))
     }
 
-    /// Specialized version of `merge`.
-    pub fn merge2<T1,T2,T:Data>(&self, label:Label, t1:&T1, t2:&T2) -> Stream<T>
+    /// Specialized version of `any`.
+    pub fn any2<T1,T2,T:Data>(&self, label:Label, t1:&T1, t2:&T2) -> Stream<T>
     where T1:EventOutput<Output=T>, T2:EventOutput<Output=T> {
-        self.register(OwnedMerge::new2(label,t1,t2))
+        self.register(OwnedAny::new2(label,t1,t2))
     }
 
-    /// Specialized version of `merge`.
-    pub fn merge3<T1,T2,T3,T:Data>(&self, label:Label, t1:&T1, t2:&T2, t3:&T3) -> Stream<T>
+    /// Specialized version of `any`.
+    pub fn any3<T1,T2,T3,T:Data>(&self, label:Label, t1:&T1, t2:&T2, t3:&T3) -> Stream<T>
     where T1:EventOutput<Output=T>, T2:EventOutput<Output=T>, T3:EventOutput<Output=T> {
-        self.register(OwnedMerge::new3(label,t1,t2,t3))
+        self.register(OwnedAny::new3(label,t1,t2,t3))
     }
 
-    /// Specialized version of `merge`.
-    pub fn merge4<T1,T2,T3,T4,T:Data>
+    /// Specialized version of `any`.
+    pub fn any4<T1,T2,T3,T4,T:Data>
     (&self, label:Label, t1:&T1, t2:&T2, t3:&T3, t4:&T4) -> Stream<T>
     where T1:EventOutput<Output=T>,
           T2:EventOutput<Output=T>,
           T3:EventOutput<Output=T>,
           T4:EventOutput<Output=T> {
-        self.register(OwnedMerge::new4(label,t1,t2,t3,t4))
+        self.register(OwnedAny::new4(label,t1,t2,t3,t4))
     }
 
 
-    // === Merge_ ===
+    // === Any_ ===
 
-    /// Like `gather` but drops the incoming data. You can attach streams of different types.
-    pub fn gather_(&self, label:Label) -> Merge_ {
-        self.register_raw(OwnedMerge_::new(label))
+    /// Like `any_mut` but drops the incoming data. You can attach streams of different types.
+    pub fn any_mut_(&self, label:Label) -> Any_ {
```

Comment:
Can we rename it to any_mut_dropping? There seems to be a convention for finishing with _ things which are dropping stuff, but:
1. This is Yen Another Convention You Must Learn To Contribute
2. I believe we have functions which end with _ without any dropping semantic.

Reply:
I'm against this change. `_` means drop both in `_ = ...` as well as in pattern matching, etc. Moreover there are libraries which already use this style in Rust. And its more consistent.